minorminer/utils/feasibility.py

```diff
@@ -0,0 +1,222 @@
+            embeddings.
+
+    Raises:
+        ValueError: If `T` and `topology` are inconsistent, or if `T` and `t` are inconsistent.
+        ValueError: If neither `T` nor `topology` is specified.
+        ValueError: If `S` cannot be embedded in `T` or the specified topology.
+
+    Returns:
+        float: Minimum `lattice_size` for embedding to be feasible. Returns
+            None if embedding for any number of graph rows is infeasible.
+    """
+    # Comment, similar bounds are possible allowing for minor embeddings,
+    # the degree distribution is bounded as chain length increases, whilst
+    # the number of nodes is depleted.
+    # This could be a possible feature expansion.
+    if T is not None:
+        if embedding_feasibility_filter(S, T, one_to_one) is False:
+            return None
+        if topology is None:
+            topology = T.graph.get("family")
+        elif topology != T.graph.get("family"):
+            raise ValueError("Arguments T and topology are inconsistent")
+        if t is None:
+            t = T.graph["tile"]
+    else:
+        if topology is None:
+            raise ValueError("T or topology must be specified")
+        if t is None:
+            t = 4
+        max_degrees = {"chimera": 2 + 2 * t, "pegasus": 15, "zephyr": 4 + 4 * t}
+        max_source_degree = max(S.degree[n] for n in S.nodes())
+        if max_source_degree > max_degrees[topology]:
+            return None
+
+    N = S.number_of_nodes()
+    if topology == "chimera":
+        # Two colorability is necessary, and cheap to check
+        if any(
+            c > 2
+            for c in nx.algorithms.coloring.greedy_coloring.greedy_color(
+                S, strategy="connected_sequential_bfs"
+            ).values()
+        ):
+            return None
+
+        def generator(lattice_size):
+            return dnx.chimera_graph(m=lattice_size, n=lattice_size, t=t)
+
+        # A lower bound based on number of variables N = m*n*2*t
+        lattice_size = np.ceil(np.sqrt(N / 4 / t))
+    elif topology == "pegasus":
+
+        def generator(lattice_size):
+            return dnx.pegasus_graph(m=lattice_size)
+
+        # A lower bound based on number of variables N = (m*24-8)*(m-1)
+        lattice_size = np.ceil(1 / 12 * (8 + np.sqrt(6 * N + 16)))
+    elif topology == "zephyr":
+
+        def generator(lattice_size):
+            return dnx.zephyr_graph(m=lattice_size, t=t)
+
+        # A lower bound based on number of variables N = (2m+1)*m*4*t
+        lattice_size = np.ceil((np.sqrt(2 * N / t + 1) - 1) / 4)
+    else:
+        raise ValueError(
+            "source graphs must be a graph constructed by "
+            "dwave_networkx as chimera, pegasus or zephyr type"
+        )
+    # Evaluate tile feasibility (defect free subgraphs)
+    lattice_size = round(lattice_size)
+    tile = generator(lattice_size=lattice_size)
+    while embedding_feasibility_filter(S, tile, one_to_one) is False:
```

Comment:
I presume when the lattice-size becomes large enough, `embedding_feasibility_filter(S, tile, one_to_one)` will always turn True? Maybe add a comment here to clarify that this loop will always end at some point.

Reply:
PEP-8 recommends to [avoid boolean value comparison](https://peps.python.org/pep-0008/#:~:text=Don't%20compare%20boolean%20values) with `==` or `is`:
```suggestion
    while not embedding_feasibility_filter(S, tile, one_to_one):
```